hazelcast/src/test/java/com/hazelcast/cp/internal/datastructures/lock/FencedLockBasicTest.java

```diff
@@ -103,4 +109,46 @@ public void test_lockFailsAfterCPGroupDestroyed() throws ExecutionException, Int
 
         proxyInstance.getCPSubsystem().getLock(lock.getName());
     }
+
+    @Test(timeout = 60000)
+    public void testDestroy_RemovesLockProxy() {
+        String lockName = UUID.randomUUID().toString();
+        FencedLock myLock = instances[0].getCPSubsystem().getLock(lockName);
+        myLock.lock();
+        try {
+        } finally {
+            myLock.unlock();
+        }
+        myLock.destroy();
+
+        assertTrueEventually(() -> {
+            LockService service = getNodeEngineImpl(instances[0]).getService(LockService.SERVICE_NAME);
+            ConcurrentMap<String, FencedLockProxy> proxies = ReflectionUtils.getFieldValueReflectively(service, "proxies");
+            assertFalse(proxies.containsKey(lockName));
```

Comment:
Since we can have confusion between Lock's name and Lock's fully qualified name, maybe it will be better to check that the `proxies` map is empty:
`assertEquals(0, proxies.size());`

Reply:
Yes, I think it would be better to move this test to another test class (or a separate class) since it will also be executed during `FencedLockClientBasicTest` (no sense to run) and `FencedLockCompatibilityTest` (can fail).

Reply:
Will move it.